### PR TITLE
[SPARK-46015][DOCS][PS] Remove broken link for Koalas issues

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to remove broken link for Koalas issues.

### Why are the changes needed?

To remove the broken - also out dated - link.

### Does this PR introduce _any_ user-facing change?

No, docs only.

### How was this patch tested?

The existing CI should pass.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: python/docs/source/user_guide/pandas_on_spark/transform_apply.rst [hunk 1]
```diff
@@ -20,7 +20,7 @@
 Transform and apply a function
 ==============================
 
-.. NOTE: the images are stored at https://github.com/koalas/issues/1443. Feel free to edit and/or add.
+.. NOTE: the images are stored at https://github.com/databricks/koalas/issues/1443. Feel free to edit and/or add.
```
Comment (itholic) [inline python/docs/source/user_guide/pandas_on_spark/transform_apply.rst:23]: On my second thought, maybe we can just remove it?

Comment (dongjoon-hyun) [reply python/docs/source/user_guide/pandas_on_spark/transform_apply.rst:23]: Yes, let's remove it, @itholic . :)

Comment (itholic) [reply python/docs/source/user_guide/pandas_on_spark/transform_apply.rst:23]: cc @HyukjinKwon FYI as an original author.

**Review by HyukjinKwon — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by zhengruifeng — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**